echo-core/src/main/java/com/netflix/spinnaker/echo/services/IgorService.java

```diff
@@ -30,6 +31,12 @@ Map<String, Object> getBuild(
       @Path("master") String master,
       @Path(value = "job", encode = false) String job);
 
+  @GET("/builds/status/{buildNumber}/{master}")
+  Map<String, Object> getBuildStatusWithJobQueryParameter(
+      @NotNull @Path("buildNumber") Integer buildNumber,
+      @NotNull @Path("master") String master,
+      @NotNull @Query(value = "job") String job);
```

Comment:
ONLY call out is... encoding. Calling out the caution on the other PR, but ... not sure how much of an issue that really is.  IF job should be URL encoded due to any kinda characters - I don't THINK so at this point though.